### PR TITLE
feat: use the casbin model to store relationships between users and groups

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
The pr use casbin store the relationship tetween users and groups, replace the old query statment that using `like` to query users in group.  Note that this commit now does not consider migrating existing user and group relationships into casbin, which would result in users not being found in the organization's group management. I'm not sure if I need to add a migration file before the pr be merged.

The `GroupManager` is not being used now. However, it may then be used to store group hierarchy to determine the inherited permissions of all groups above the user.

Next step, improve the permisson page, support UI to add permission for a group. Because of the implement of using casbin manager user in Casdoor, then we can support the more convenient API, such as `HasPermissionForUser()` to ckeck if a user(group) has permision. It will be more friendly than using Enforce API directly.

Fix: https://github.com/casdoor/casdoor/issues/2020

### CODE REVIEW

File: object/adapter.go [hunk 1]
```diff
@@ -181,7 +185,8 @@ func (adapter *Adapter) initAdapter() error {
 		}
 
 		var err error
-		adapter.Adapter, err = xormadapter.NewAdapterByEngineWithTableName(NewAdapter(adapter.DatabaseType, dataSourceName, adapter.Database).Engine, adapter.getTable(), adapter.TableNamePrefix)
+		engine, err := xorm.NewEngine(adapter.DatabaseType, dataSourceName)
```
Comment (leo220yuyaodog) [inline object/adapter.go:188]: If use `NewAdapter()` to crete connect, it will cause `sql: connect is closed error`.  Because it use the same connect as the ormer.

File: object/user_enforcer.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package object
+
+import (
+	"github.com/casbin/casbin/v2"
+	"github.com/casbin/casbin/v2/errors"
+	"github.com/casdoor/casdoor/util"
+)
+
+type IUserGroupEnforcer interface {
```
Comment (hsluoyz) [inline object/user_enforcer.go:9]: No interface

File: controllers/user.go [hunk 3]
```diff
@@ -90,7 +90,7 @@ func (c *ApiController) GetUsers() {
 
 	if limit == "" || page == "" {
 		if groupName != "" {
-			maskedUsers, err := object.GetMaskedUsers(object.GetGroupUsers(groupName))
+			maskedUsers, err := object.GetMaskedUsers(object.GetGroupUsers(util.GetId(owner, groupName)))
```
Comment (hsluoyz) [inline controllers/user.go:93]: Isn't this already merged?

File: object/adapter.go [hunk 4]
```diff
@@ -313,7 +318,7 @@ func (adapter *Adapter) builtInAdapter() bool {
 		return false
 	}
 
-	return adapter.Name == "permission-adapter-built-in" || adapter.Name == "api-adapter-built-in"
+	return adapter.Name == "user-adapter-built-in" || adapter.Name == "api-adapter-built-in"
```
Comment (hsluoyz) [inline object/adapter.go:321]: Isn't this already merged?

File: object/enforcer.go [hunk 5]
```diff
@@ -121,6 +121,10 @@ func DeleteEnforcer(enforcer *Enforcer) (bool, error) {
 }
 
 func (enforcer *Enforcer) InitEnforcer() error {
+	if enforcer == nil {
```
Comment (hsluoyz) [inline object/enforcer.go:124]: Isn't this already merged?

File: object/group.go [hunk 6]
```diff
@@ -214,39 +214,33 @@ func ConvertToTreeData(groups []*Group, parentId string) []*Group {
...
 	users := []*User{}
+	owner, _ := util.GetOwnerAndNameFromId(groupId)
+	names, err := userEnforcer.GetUserNamesByGroupName(groupId)
+	if err != nil {
+		return nil, err
+	}
+
 	session := ormer.Engine.Table("user").
-		Where(builder.Like{"`groups`", groupId + "\""})
+		Where("owner = ?", owner).In("name", names)
```
Comment (hsluoyz) [inline object/group.go:243]: Isn't this already merged?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/group.go [hunk 7]
```diff
@@ -265,7 +259,7 @@ func GetPaginationGroupUsers(groupId string, offset, limit int, field, value, so
 		session = session.Desc(fmt.Sprintf("user.%s", util.SnakeString(sortField)))
 	}
 
-	err := session.Find(&users)
+	err = session.Find(&users)
```
Comment (hsluoyz) [inline object/group.go:262]: Why change this? Is this related to the goal of this PR? If not, then discard it

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: object/group.go (hunk 7 above)
Comment (leo220yuyaodog) [reply object/group.go:262]: Because the added code above has defined err, use `:=` cannot compile.

File: controllers/user.go (hunk 3 above)
Comment (leo220yuyaodog) [reply controllers/user.go:93]: No, has rebased

**Review by hsluoyz — APPROVED** [review]
